### DISCUSSION TITLE
Portrayal shapes in mesa-geo

### DISCUSSION THREAD

**Discussion by joshuashakya** [mesa/mesa-geo#64] [Q&A]
I am using mesa-geo for a project but am unable to change the shape of the GeoAgent using portrayal, only "circle" shape works.
I would like to get a rectangle:

    portrayal["Shape"] = "rect"
    portrayal["w"] = 2
    portrayal["h"] = 2

Does anybody have a suggestion regarding this ?

**Comment by rht** [comment]
Looks like only circle is supported in `mesa-geo` for now: https://github.com/projectmesa/mesa-geo/blob/8e556556191438bda7db9d1fd11c08867aae9a7e/mesa_geo/visualization/templates/js/MapModule.js#L26.
This is the corresponding grid drawer in Mesa: https://github.com/projectmesa/mesa/blob/def2c1be78720632937014900659d480f972b023/mesa/visualization/templates/js/GridDraw.js#L82-L120.

**Comment by wang-boyu** [comment]
Leaflet.js supports the drawing of rectangles as a special kind of polygon: https://leafletjs.com/reference-1.7.1.html#rectangle

What we can probably do in mesa-geo is

1. In `MapModule.js`, create `L.rectangle` instead of `L.circleMarker` when users specify `portrayal["shape"] = "rect"`.
2. Calculate the `bounds` of rectangle based on the `portrayal["w"]`, `portrayal["h"]` and `geoagent.centroid` values.
3. Populate other styling properties such as color, weight, etc from `portrayal`.
